artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java

```diff
@@ -656,6 +656,7 @@ public void removeItself() throws Exception {
          browserDeliverer.close();
       } else {
          messageQueue.removeConsumer(this);
+         messageQueue.deliverAsync();
```

Comment:
Kind of wonder if it shouldnt be the queue doing this instead of the consumer? Also feels like it may be a separate change really.

Reply:
it could be but the test that now has an exclusive and non exclusive variant was  intermittently failing and this was the reason.

Reply:
sure, but that seems more like a bug in the queue itself, and this change isnt specific to openwire, should have been addressed seperately from this openwire-consumer-specific JIRA.